### ISSUE TITLE
atom --wait works only after closing the window, not just the file tab

### ISSUE THREAD

**Issue by raduserbanescu** [atom/atom#19849]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: <https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode>
    * Followed all applicable steps in the debugging guide: <https://flight-manual.atom.io/hacking-atom/sections/debugging/>
    * Checked the FAQs on the message board for common solutions: <https://discuss.atom.io/c/faq>
    * Checked that your issue isn't already filed: <https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom>
    * Checked that there is not already an Atom package that provides the described functionality: <https://atom.io/packages>

### Description

I have always had `git config core.editor` set to `atom --wait`.
_NOTE:_ This also happens if I set the config to `atom --wait --safe`.

This issue appeared after I upgraded from version **1.36.0** to **1.40.1**

An old closed issue that might be related to this one: #16740

### Steps to Reproduce

1. Open the project folder in Atom.
2. Edit files and run `git commit`.
3. Because the project folder is already opened in Atom, the `COMMIT_EDITMSG` file is opened in a new tab in the existing window.
4. Write message an save.

Of course this also happens for `git rebase -i` and any other git operations that require input from the editor.

**Expected behavior:**

After closing the file tab the commit should finalize.

**Actual behavior:**

You need to close the window for the commit to finalize.

**Reproduces how often:**

100% of the time.

### Versions

```
Atom    : 1.40.1
Electron: 3.1.10
Chrome  : 66.0.3359.181
Node    : 10.2.0

git version 2.21.0.windows.1

Windows 10 Enterprise 64bit
```

### Additional Information

Demo:

![atom--wait_issue](https://user-images.githubusercontent.com/3825876/63950139-1fd41f80-ca84-11e9-99ae-b66a191ca56d.gif)

**Comment by calebmeyer** [comment]
This is inconsistently reproducing for me (also win10 enterprise). The first time I tried it, the `atom --wait` command, it behaved as you've shown. Closing the file had no effect, closing the editor returned from the wait command.

Opening the file a second time (in an existing window) with the same `atom --wait README.md` worked as expected. As soon as I closed the file, the wait command returned.

Closing all my atom windows and opening the file a third and fourth time once again produced the correct behavior.

I did see a message on my terminal after the prompt had been printed again:
```
 Attempting to call a function in a renderer window that has been closed or released.
Function provided here: Object.<anonymous> (C:\Users\CM022291\AppData\Local\atom\app-1.40.1\resources\app.asar\node_modules\github\lib\worker.js:79:22
Remote event names: destroyed, crashed
```

**Comment by ScienceofSpock** [comment]
I have the same issue on 2 machines. My laptop running Win 10 home and my desktop running win 10 pro. Using Atom 1.40.1, git version 2.23.0.windows.1. On both machines, this behavior happens 100% of the time.

**Comment by wadethestealth** [comment]
I too have experienced this for quite a while now. I believe the problem is that when using atom as a git editor, it calls `atom --wait` which only responds when the entire window is closed. However `atom --wait file` will respond when the corresponding file closes. Not sure if that makes this a bug, but it is definitely an issue an a weird workflow especially since the git editor with just open the previous workspace for the directory or adds it to my open atom window of the directory.

**Comment by ScienceofSpock** [comment]
@wadethestealth I tried the "atom --wait file" option and all that does is open a new blank file named "file" alongside the commit message. It still hangs and waits for the entire app to close.

**Comment by wadethestealth** [comment]
@ScienceofSpock you misunderstood me. File is not literal it's a name of a file you want to wait on being closed. Git must supply the file name not you. If you want to test the behavior open your terminal and type `atom --wait untitled.txt` then type something save and close that file all in atom. Your terminal should not allow you to do anything until the file closes. But you should be able to leave the atom window open.

**Comment by ScienceofSpock** [comment]
@wadethestealth yes yes, I know how it works, however, your post made it sound like the additional option would force the proper behavior, so I tried it. Still doesn't work. Git stays waiting for the file to close, even though the file IS closed, and will only complete if I actually close the atom editor completely.

**Comment by wadethestealth** [comment]
Regardless it's not working for me normally either though.

**Comment by krzysztof-sikorski** [comment]
This bug always triggers on my Windows installation, and so far I only found one workaround: use `atom --wait --new-window` instead of `atom --wait`.

This will replicate Atom's original behaviour of opening the commit message in a fresh new window instead of adding a tab to an existing window - this way I only have to close that new window (which is as simple as hitting `CTRL+W` twice) instead of having to close and then reopen a project window.

**Comment by christos-stamatis** [comment]
As @krzysztof-sikorski mentioned the command atom --wait --new-window will do the trick. It seems like git somehow connects with the Atom window that is active (selected) and requires that window to be closed entirely rather than the tab. 

Versions:

Atom    : 1.44.0
Electron: 4.2.7
Chrome  : 69.0.3497.128
Node    : 10.11.0

git version 2.25.0.windows.1

Windows 10 Pro 64bit
